File: 02_ProgrammingFundamentals/_05_Lists/_03_MoreExercises/_04_MixedUpLists.java
package _02_ProgrammingFundamentals._05_Lists._03_MoreExercises;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;
import java.util.stream.Collectors;

public class _04_MixedUpLists {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        List<Integer> numbersListOne = Arrays.stream(scan.nextLine().split("\\s+")).map(Integer::parseInt).collect(Collectors.toList());
        List<Integer> numbersListTwo = Arrays.stream(scan.nextLine().split("\\s+")).map(Integer::parseInt).collect(Collectors.toList());

        // Add variable
        int[] numbers = new int[2];

        // Determine remaining elements
        if (numbersListOne.size() > numbersListTwo.size()) {
            numbers[0] = numbersListOne.get(numbersListOne.size() - 1);
            numbers[1] = numbersListOne.get(numbersListOne.size() - 2);

            numbersListOne.remove(numbersListOne.size() - 1);
            numbersListOne.remove(numbersListOne.size() - 1);
        } else {
            numbers[0] = numbersListTwo.get(0);
            numbers[1] = numbersListTwo.get(1);

            numbersListTwo.remove(0);
            numbersListTwo.remove(1);
        }

        numbersListOne.addAll(numbersListTwo);
        numbersListTwo.clear();

        // Sort range elements
        Arrays.sort(numbers);

        // Check and save elements that fit the requirement
        for (int i = 0; i < numbersListOne.size(); i++) {
            if (numbersListOne.get(i) > numbers[0] && numbersListOne.get(i) < numbers[1]) {
                numbersListTwo.add(numbersListOne.get(i));
            }
        }

        Collections.sort(numbersListTwo);

        // Print result
        numbersListTwo.stream().forEach(e -> System.out.print(e + " "));

        scan.close();
    }
}

File: 24_C#Fundamentals/_05_Lists/_03_MoreExercises/_04_MixedUpLists/_04_MixedUpLists.cs
using System;
using System.Xml.Linq;

public class _04_MixedUpLists
{
    public static void Main(string[] args)
    {
        // Read input values
        List<int> numbersListOne = Console.ReadLine().Split(" ").Select(int.Parse).ToList();
        List<int> numbersListTwo = Console.ReadLine().Split(" ").Select(int.Parse).ToList();

        // Add variable
        int[] numbers = new int[2];

        // Determine remaining elements
        if (numbersListOne.Count > numbersListTwo.Count)
        {
            numbers[0] = numbersListOne[numbersListOne.Count - 1];
            numbers[1] = numbersListOne[numbersListOne.Count - 2];

            numbersListOne.RemoveAt(numbersListOne.Count - 1);
            numbersListOne.RemoveAt(numbersListOne.Count - 1);
        }
        else
        {
            numbers[0] = numbersListTwo[0];
            numbers[1] = numbersListTwo[1];

            numbersListTwo.RemoveAt(0);
            numbersListTwo.RemoveAt(1);
        }

        numbersListOne.AddRange(numbersListTwo);
        numbersListTwo.Clear();

        // Sort range elements
        Array.Sort(numbers);

        // Check and save elements that fit the requirement
        foreach (int num in numbersListOne)
        {
            if (num > numbers[0] && num < numbers[1])
            {
                numbersListTwo.Add(num);
            }
        }


        numbersListTwo.Sort();

        // Print result
        Console.WriteLine(string.Join(" ", numbersListTwo));
    }
}
